app/controllers/carto/api/public/federated_tables_controller.rb

```
@@ -226,6 +248,25 @@ def render_paged(result, total)
 
           render_jsonp(enriched_response, 200)
         end
+
+        def handle_service_error(exception)
+          message = get_error_message(exception)
+          case message
+          when /Server (.*) does not exist/
+            raise Carto::LoadError.new(message)
+          when /Not enough permissions to access the server (.*)/
+            raise Carto::UnauthorizedError.new(message)
+          when /Could not import table (.*) of server (.*)/, /Could not import table (.*) as (.*) already exists/, /non integer id_column (.*)/, /non geometry column (.*)/
+            raise Carto::UnprocesableEntityError.new(message)
+          else
+            raise exception
```

Are you sure we want to raise an exception here? I'd rather return a 422 with the message as default (covering the previous errors with it too).

---

We cannot wrap every error as a 422. For instance, if there was a connection issue we cannot blame the user for that, 4xx means that the user did something wrong. 

Re-throwing the original exception here we'll respond with 500, and the original error message which is more accurate. It's done by rails by default.

---

A connection issue IS a user error and we should blame them; we are connecting to their service.

---

Well, we can't assume that always, even in many cases:
- Issues in DNS resolution.
- Network gitch.
- Timeouts somewhere else: e.g.: proxies.
- Whatever you can imagine: DB servers not allowed to make HTTP calls to the wild internet by configuration.

A 500 error code is a catch-all response, used when we cannot find a better error code.